workbench-core/authorization/src/dynamicAuthorization/wbcGroupManagementPlugin.ts

```diff
@@ -111,10 +111,17 @@ export class WBCGroupManagementPlugin implements GroupManagementPlugin {
   public async addUserToGroup(request: AddUserToGroupRequest): Promise<AddUserToGroupResponse> {
     const { groupId, userId } = request;
 
-    // ToDo: This requires a check to ensure status of group isn't in pending_delete
     // ToDo: Audit authenticatedUser which actor is performing operation
 
     try {
+      const {
+        data: { status }
+      } = await this.getGroupStatus({ groupId });
+
+      if (status === 'delete_pending') {
+        throw new ForbiddenError(`Cannot assign user to group '${groupId}'. It is pending delete.`);
```

Comment:
Would a `GroupNotFoundError` make more sense here since the group is in the process of deletion?

Reply:
I used `ForbiddenError` to be consistent with `setGroupStatus` to indicate that operation is not allowed when group is in invalid state. Should there be  another type (e.g. `InvalidGroupStateError`) to indicate this precisely?

Reply:
The `ForbiddenError` there was because changing the status from `delete_pending` to `active` is not allowed. `GroupNotFoundError` makes sense to me because the group doesnt exist anymore, as it is currently being deleted